open-machine-learning-jupyter-book/deep-learning/GAN.md

```diff
@@ -13,5 +13,274 @@ kernelspec:
   name: python3
 ---
 
-# GAN (TBD)
+# Generative adversarial networks 
 
+The original purpose of GAN is to generate new data. It classically generates new images, but is applicable to wide range of domains.
+It learns the training set distribution and can generate new images that have never been seen before.
+In contrast to e.g., autoregressive models or RNNs (generating one word at a time), GANs generate the whole output all at once.
+
+```{seealso}
+GAN is proposed in 2005, the paper is [Generative Adversarial Networks](https://arxiv.org/abs/1406.2661).
+```
+
+## overview
+
+The structure of GAN is roughly as follows. It has two essential part, discriminator and generator. 
+Discriminator learns to become better at distinguishing real from generated images, and generator learns to generate better images to fool the discriminator.
+
+:::{figure-md} GAN structure
+<img src="../../images/deep-learning/GAN/structure.png" width="90%" class="bg-white mb-1">
+
+Illustration of GAN structure
+:::
+
+Then, one question appears: why Are GANs Are Called Generative Models?
+-The generative part comes from the fact that the model "generates" new data,
+- Usually, generative models use an approximation to compute the usually intractable distribution; here, the discriminator part does that approximation,
+- So, it does learn p(x).
+
+## Training
+
+To make the model converge, we need to find a proper objective for it.
+For the GAN, it is $\min\limits_{G} \max\limits_{D} V(D, G) = \mathbb{E}_{x \thicksim p_{data}}[log D(x)] + \mathbb{E}_{z\thicksim p_z(z)}[log(1-D(G(z)))]$
+
+### Discriminator gradient for update (gradient ascent)
+
+The aim of Discriminator is $\bigtriangledown_{W_D} \frac{1}{n} \sum_{i=1}^n [log D(x^{(i)}) + log (1-D(G(z^{(i)})))]$. We can split it into two parts.
+First is $D(x^{(i)})$. If it predicts well in real images, the probability will be close to 1. 
+Second is $D(G(z^{(i)}))$. This part use to predict the fake images which generdated from the generator, and if it predicts well, the probability will be close to 0.
+
+### Generator gradient for update (gradient descent)
+
+The aim of Generator is $\bigtriangledown_{W_G} \frac{1}{n} \sum_{i=1}^n log(1-D(G(z^{(i)})))$. 
+$D(G(z^{(i)}))$ is also used to predict fake images, but if it predicts badly on fake images, the probability will be close to 1, which is not same to discriminator.
+
+### Algorithm
+
+:::{figure-md} GAN algorithm
+<img src="../../images/deep-learning/GAN/algorithm.png" width="90%" class="bg-white mb-1">

```

Comment:
We aim to be math-light, so such algorithms should not be included.

Reply:
Done.